vsphere/resource_vsphere_host.go

```diff
@@ -0,0 +1,683 @@
+			},
+			"force": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Force add the host to vsphere, even if it's already managed by a different vSphere instance.",
+				Default:     false,
+			},
+			"connected": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Set the state of the host. If set to false then the host will be asked to disconnect.",
+				Default:     true,
+			},
+			"maintenance": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Set the host's maintenance mode. Default is false",
+				Default:     false,
+			},
+			"lockdown": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Set the host's lockdown status. Default is disabled. Valid options are 'disabled', 'normal', 'strict'",
+				Default:     "disabled",
+			},
+		},
+	}
+}
+
+func resourceVsphereHostRead(d *schema.ResourceData, meta interface{}) error {
+
+	// NOTE: Destroying the host without telling vsphere about it will result in us not
+	// knowing that the host does not exist any more.
+
+	// Look for host
+	client := meta.(*VSphereClient).vimClient
+	hostID := d.Id()
+	if hostID == "" {
+		return nil
+	}
+
+	// Find host and get reference to it.
+	hs, err := hostsystem.FromID(client, hostID)
+	if err != nil {
+		if soap.IsSoapFault(err) {
+			sf := soap.ToSoapFault(err)
+			_, ok := sf.Detail.Fault.(types.ManagedObjectNotFound)
+			if !ok {
+				log.Printf("[DEBUG] Error while searching host %s. Error: %s", hostID, err)
+				return err
+			}
+			d.SetId("")
+			return nil
+		}
+		log.Printf("[DEBUG] non SOAP error while searching host %s. Error %#v", hostID, err)
+		return err
+	}
+
+	maintenanceState, err := hostsystem.HostInMaintenance(hs)
+	if err != nil {
+		return err
+	}
+	d.Set("maintenance", maintenanceState)
+
+	// Retrieve host's properties.
+	log.Printf("[DEBUG] Got host %#v", hs)
+	host, err := hostsystem.Properties(hs)
+	if err != nil {
+		log.Printf("[DEBUG] Error while retrieving properties for host %s. Error: %s", hostID, err)
+		return err
+	}
+
+	if host.Parent != nil {
+		d.Set("cluster", host.Parent.Value)
+	} else {
+		d.Set("cluster", "")
+	}
+
+	connectionState, err := hostsystem.GetConnectionState(hs)
+	if err != nil {
+		return err
+	}
+
+	if connectionState == types.HostSystemConnectionStateDisconnected {
+		// Config and LicenseManager cannot be used while the host is
+		// disconnected.
+		d.Set("connected", false)
+		return nil
+	}
+	d.Set("connected", true)
+
+	lockdownMode, err := hostLockdownString(host.Config.LockdownMode)
+	if err != nil {
+		return nil
+	}
+
+	log.Printf("Setting lockdown to %s", lockdownMode)
+	d.Set("lockdown", lockdownMode)
+
+	lm := license.NewManager(client.Client)
+	am, err := lm.AssignmentManager(context.TODO())
+	if err != nil {
+		return err
+	}
+	licenses, err := am.QueryAssigned(context.TODO(), hostID)
+	if err != nil {
+		return err
+	}
+
+	licenseKey := d.Get("license").(string)
```

Comment:
If no license is currently known, then the `Read` function won't set it in the state here. Is that intentional?

Reply:
I do set the license to an empty string in line 181 hoping that it will prompt the user to update the config and set a proper licence. Can you suggest an alternative solution ?